src/shared.ts

```diff
@@ -106,16 +109,43 @@ export const loadScript = (): Promise<StripeConnectWrapper | null> => {
 };
 
 export const initStripeConnect = (
-  stripeConnectPromise: StripeConnectWrapper | null
-): any | null => {
-  if (stripeConnectPromise === null) {
-    return null;
-  }
+  stripePromise: Promise<StripeConnectWrapper | null>,
+  initParams: IStripeConnectInitParams
+): StripeConnectInstance => {
+  const stripeConnectInstance = stripePromise.then((wrapper) =>
+    wrapper?.initialize(initParams)
+  );
+  return {
+    create: (tagName) => {
+      let htmlName = componentNameMapping[tagName];
+      if (!htmlName) {
+        htmlName = tagName as ConnectElementHTMLName;
+      }
+      const element = document.createElement(htmlName);
+      stripeConnectInstance.then((instance) => {
+        (element as any).setConnector((instance as any).connect);
+      });
 
-  return stripeConnectPromise;
+      return element;
+    },
+    update: (updateOptions) => {
+      stripeConnectInstance.then((instance) => {
+        instance?.update(updateOptions);
+      });
+    },
+    debugInstance: () => {
+      return stripeConnectInstance;
+    },
+    logout: () => {
+      return stripeConnectInstance.then((instance) => {
+        instance?.logout();
+      });
+    },
+  };
 };
 
 const createWrapper = (stripeConnect: any) => {
+  (window as any).StripeConnect.optimizedLoading = true;
```

Comment:
This code should work if `window.StripeConnect` is not defined! I believe you need:

```suggestion
  (window as any).StripeConnect = (window as any).StripeConnect || {};
  (window as any).StripeConnect.optimizedLoading = true;
```